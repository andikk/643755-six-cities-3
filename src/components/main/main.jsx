import React from "react";
import PropTypes from "prop-types";
import PlacesList from "../places-list/places-list.jsx";
import PlacesSorting from "../places-sorting/places-sorting.jsx";
import Map from "../map/map.jsx";
import {connect} from "react-redux";
import CitiesList from "../cities-list/cities-list.jsx";
import {getCitiesListSelector, getCitySelector, getSortedOffersInCitySelector, getCoordinatesInCitySelector, getOffersInCitySelector} from "../../selectors.js";
import {ActionCreator} from "../../reducer";

// главная страница
const Main = (props) => {
  const {offers, onHeaderClick, city, citiesList, onCityClick, coordinates, onFilterClick, activeFilter, onCardHover, activeOffer} = props;
  const offersCount = offers.length;
  console.log(coordinates);

  return (
    <div className="page page--gray page--main">
      <header className="header">
        <div className="container">
          <div className="header__wrapper">
            <div className="header__left">
              <a className="header__logo-link header__logo-link--active">
                <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41"/>
              </a>
            </div>
            <nav className="header__nav">
              <ul className="header__nav-list">
                <li className="header__nav-item user">
                  <a className="header__nav-link header__nav-link--profile" href="#">
                    <div className="header__avatar-wrapper user__avatar-wrapper">
                    </div>
                    <span className="header__user-name user__name">Oliver.conner@gmail.com</span>
                  </a>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </header>

      <main className={`page__main page__main--index ${(offersCount === 0) ? ` page__main--index-empty` : ``}`}>
        <h1 className="visually-hidden">Cities</h1>
        <CitiesList citiesList={citiesList} city={city} onCityClick={onCityClick}/>
        <div className="cities">
          {(offersCount > 1) &&
            <div className="cities__places-container container">
              <section className="cities__places places">
                <h2 className="visually-hidden">Places</h2>
                <b className="places__found">{offersCount} places to stay in {city}</b>

                <PlacesSorting onFilterClick={onFilterClick} activeFilter={activeFilter}/>

                <PlacesList className="cities__places-list tabs__content" offers={offers}
                  onHeaderClick={onHeaderClick} onCardHover={onCardHover}/>
              </section>
              <div className="cities__right-section">
                <Map className={`cities__map`} coordinates={coordinates} activeMarker={activeOffer.coordinates}/>
              </div>
            </div>
          }

          {(offersCount === 0) &&
            <div className="cities__places-container cities__places-container--empty container">
              <section className="cities__no-places">
                <div className="cities__status-wrapper tabs__content">
                  <b className="cities__status">No places to stay available</b>
                  <p className="cities__status-description">We could not find any property availbale at the moment in Dusseldorf</p>
                </div>
              </section>
              <div className="cities__right-section"></div>
            </div>
          }

        </div>
      </main>
    </div>
  );
};

Main.propTypes = {
  offers: PropTypes.array.isRequired,
  city: PropTypes.string.isRequired,
  citiesList: PropTypes.array.isRequired,
  onHeaderClick: PropTypes.func.isRequired,
  onCityClick: PropTypes.func.isRequired,
  coordinates: PropTypes.array.isRequired,
  onFilterClick: PropTypes.func.isRequired,
  activeFilter: PropTypes.string.isRequired,
  onCardHover: PropTypes.func,
  activeOffer: PropTypes.object
};

Main.defaultProps = {
  onHeaderClick: () => {}
};

const mapStateToProps = (state) => {
  return {
    offers: getSortedOffersInCitySelector(state),
    city: getCitySelector(state),
    activeFilter: state.activeFilter,
    citiesList: getCitiesListSelector(state),
    coordinates: getCoordinatesInCitySelector(state),
    activeOffer: state.activeOffer
  };
};

const mapDispatchToProps = (dispatch) => ({
  onCityClick: (evt, city) => {
    evt.preventDefault();
    dispatch(ActionCreator.setCity(city));
  },
  onFilterClick: (activeFilter) => {
    dispatch(ActionCreator.setFilter(activeFilter));
  },
  onCardHover: (activeOffer) => {
    dispatch(ActionCreator.setActiveOffer(activeOffer));
  }
});

export {Main};
export default connect(mapStateToProps, mapDispatchToProps)(Main);
