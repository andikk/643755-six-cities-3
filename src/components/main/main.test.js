import React from "react";
import renderer from "react-test-renderer";
import {Main} from "./main.jsx";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router-dom";

const mockStore = configureStore([]);
const offers = [
  {id: 1,
    city: {name: `Cologne`, location: {"latitude": 50.938361, "longitude": 6.959974, "zoom": 13}},
    src: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg`,
    photos: [`https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg`],
    description: `Design interior in most sympathetic area! Complitely renovated, well-equipped, cosy studio in idyllic, over 100 years old wooden house. Calm street, fast connection to center and airport.`,
    premium: false,
    bedrooms: 5,
    guests: 5,
    features: [`Breakfast`, `Washer`, `Air conditioning`, `Laptop friendly workspace`, `Baby seat`],
    owner: {id: 25, name: `Angelina`, super: true, src: `img/avatar-angelina.jpg`},
    price: 403,
    rating: 2.9,
    name: `Wood and stone place`,
    type: `house`,
    coordinates: [50.932361, 6.937974],
    reviews: [],
    neighborhood: [],
    isFavorite: false}
];
const store = mockStore({
  city: {name: `Cologne`, location: {"latitude": 50.938361, "longitude": 6.959974, "zoom": 13}},
  offersIds: [1],
  offersMap: {
    1: {id: 1,
      city: {name: `Cologne`, location: {"latitude": 50.938361, "longitude": 6.959974, "zoom": 13}},
      src: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg`,
      photos: [`https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg`, `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg`],
      description: `Design interior in most sympathetic area! Complitely renovated, well-equipped, cosy studio in idyllic, over 100 years old wooden house. Calm street, fast connection to center and airport.`,
      premium: false,
      bedrooms: 5,
      guests: 5,
      features: [`Breakfast`, `Washer`, `Air conditioning`, `Laptop friendly workspace`, `Baby seat`],
      owner: {id: 25, name: `Angelina`, super: true, src: `img/avatar-angelina.jpg`},
      price: 403,
      rating: 2.9,
      name: `Wood and stone place`,
      type: `house`,
      coordinates: [50.932361, 6.937974],
      reviews: [],
      neighborhood: [],
      isFavorite: false}
  },
  offersNearbyIds: [],
  offersFavoritesIds: [],
  reviews: [],
  activeOfferId: null,
  authorizationStatus: `NO_AUTH`,
  user: {
    avatar: `img/1.png`,
    id: 4,
    isTrust: false,
    name: `Max`
  },
  activeFilter: {label: `Popular`, value: `ALL`}
});

it(`Should Main render correctly`, () => {

  const tree = renderer
    .create(<Provider store={store}><MemoryRouter><Main
      offers={offers}
      city={{name: `Cologne`, location: {"latitude": 50.938361, "longitude": 6.959974, "zoom": 13}}}
      citiesList={[{name: `Cologne`, location: {"latitude": 50.938361, "longitude": 6.959974, "zoom": 13}}]}
      onCityClick={()=>{}}
      coordinates={[50.938361, 6.959974]}
      onFilterClick={() => {}}
      activeFilter={{label: `Popular`, value: `ALL`}}
      onCardHover={()=>{}}
      activeOffer={{}}
      history={{}}
      onHeaderClick = {() => {}}
    /></MemoryRouter></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
