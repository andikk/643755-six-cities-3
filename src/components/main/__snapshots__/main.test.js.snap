// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Main render correctly 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                >
                   
                </div>
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index "
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item--active"
              href="#"
              onClick={[Function]}
            >
              <span>
                Cologne
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    />
  </main>
</div>
`;
